File: ExamPortalBackend/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import org.springframework.http.ResponseEntity;

import com.exam.entity.exam.Category;
import com.exam.entity.exam.Quiz;

public interface QuizService {

	// addQuiz
	public Quiz addQuiz(Quiz quiz);

	// updateQuiz
	public Quiz updateQuiz(Quiz quiz);

	// getAllQuiz
	public Set<Quiz> getAllQuiz();

	// getQuizById
	public Quiz getQuizById(int quizId);

	// deleteQuizById
	public void deleteQuizById(int quizId);

	//get Quiz by category
	public List<Quiz> getQuizzesOfCategory(Category category);
	
	//getAllActiveQuiz
	public List<Quiz> getActiveQuizzes();
	
	//getActiveQuizByCategory
	public List<Quiz> getActiveQuizByCategory(Category c);
}

File: src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http : HttpClient) { }
  
    //load all the quizes
    public quizzes()
    {
      return this._http.get(`${baseUrl}/quiz/`);
    }

    //add quiz
    public addQuiz(quiz:any)
    {
      return this._http.post(`${baseUrl}/quiz/`, quiz);
    }

    //delete quiz
    public deleteQuiz(quizid:any)
    {
      return this._http.delete(`${baseUrl}/quiz/${quizid}`, quizid);
    }

    //get a single quiz
    public getQuiz(quizid : any)
    {
      return this._http.get(`${baseUrl}/quiz/${quizid}`)
    }

    //update quiz
    public updateQuiz(quiz:any)
    {
      return this._http.put(`${baseUrl}/quiz/`, quiz);
    }

    //getQuizzesOfCategory
    public getQuizzesOfCategory(cid:any)
    {
      return this._http.get(`${baseUrl}/quiz/category/${cid}`);
    }

    //getAllActice
    public getAllActiveQuiz()
    {
      return this._http.get(`${baseUrl}/quiz/active`);
    }

    //getActiveQuizOfCategory
    public getActiveQuizOfCategory(cid:any)
    {
      return this._http.get(`${baseUrl}/quiz/category/active/${cid}`)
    }
}
